source/analyze-code.txt

```diff
@@ -0,0 +1,179 @@
+.. _visual-studio-extension-builders:
+
+=================
+Analyze Your Code
+=================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+Learn how to use the {+product+} to **statically analyze** your {+driver-long+} code.
+Static analysis is a technique where software inspects your
+code without running it.
+
+The {+product+} can analyze **builder** and **Language Integrated Query (LINQ)** expressions.
+
+A builder is a class provided by the {+driver-short+} to help you construct
+`BSON <https://bsonspec.org/>`__ objects.
+
+LINQ is a query syntax included in the C# language. The {+driver-short+}
+can translate a subset of LINQ expressions into MongoDB aggregation pipelines.
+
+To learn more about builders, see 
+`Builders <{+driver-docs+}reference/driver/definitions/>`__ in the
+{+driver-short+} documentation.
+
+To learn more about LINQ, see the following resources:
+
+- `LINQ <{+driver-docs+}reference/driver/crud/linq/>`__ in the {+driver-short+}
+  documentation
+- `LINQ <https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/>`__
+  in the Microsoft C# guide
+
+To learn more about aggregation pipelines, see
+:manual:`Aggregation </aggregation/>` in the MongoDB manual.
+
+Analyze Builders
+----------------
+
+Use the {+product+} to translate your builder expressions into the {+query-api+}.
+Click the following tabs to see an example of a builder expression
+and its corresponding {+query-api+} translation:
+
+.. tabs::
+
+   .. tab:: Builders
+      :tabid: builder
+
+      .. code-block:: csharp
+
+         var filter = Builders<Movie>.Filter.Eq(m => m.Genre, genre) &
+                      Builders<Movie>.Filter.Gte(m => m.Score, minScore) &
+                      Builders<Movie>.Filter.Regex(m => m.Score, titleSearchTerm);
+
+   .. tab:: {+query-api+}
+      :tabid: query-api-builders
+
+      .. code-block:: json
+
+         {
+           "$and": [ { "Genre": genre },
+             { "Score": { "$gte": minScore } },
+             { "Score": /titleSearchTerm/ } ]
```

Comment:
question: This surprises me as other drivers output strict json but this seems to be relaxed json. Does it output `/titleSearchTerm/` or `{ "Score": { $regex: "titleSearchTerm" } }`

Reply:
Based off of [this screenshot](https://github.com/mongodb/docs-visual-studio-extension/blob/ae895b0e9a41d26415b72de89f4db5f2f0e5dba8/source/includes/images/builder-popup.png), this MongoDB analyzer seems to output the /titleSearchTerm/ syntax.

Based off of [this documentation ](https://docs.mongodb.com/manual/reference/operator/query/regex/#mongodb-query-op.-regex)of the `$regex` operator, it seems like `{ <field>: /pattern/<options> }` is a valid way to filter against a regex expression.

I'm not sure how this JSON string is "relaxed" as opposed to "strict"? Does the proceeding JSON string contain Relaxed Extended JSON type information (I think that is the only distinguishing factor between Relaxed Extended and Strict JSON)?